### DISCUSSION TITLE
Advantages of @tauri/fs vs Node/JS fs?

### DISCUSSION THREAD

**Discussion by NikolaGrgic** [tauri-apps/tauri#2738] [Q&A] [answered]
Hi all,

I was wondering if there were any advantages to using the Tauri "fs" package vs. the built-in Node/JavaScript "fs" package for handling and dealing with files. I was hoping to adjust permissions on files and directories using `fs.chown` and `fs.chmod` and it would appear that the default Tauri package does not have that functionality.

Any suggestions?

**Comment by FabianLars** [comment, marked as answer]
The main advantage of tauris fs api is that it actually works in tauri windows lol. Tauri apps don't run in a nodejs context. It's kinda similar to trying to use nodejs in a browser.

To solve the problem you could use https://doc.rust-lang.org/std/os/unix/fs/trait.PermissionsExt.html if know rust or execute a [Command](https://tauri.studio/en/docs/api/js/classes/shell.Command) consisting of a chmod/chown call otherwise.
(it's possible that i missed a better alternative here...)
